src/frontend/src/binder/query.rs

```diff
@@ -33,19 +33,14 @@ pub struct BoundQuery {
 }
 
 impl BoundQuery {
-    /// The names returned by this [`BoundQuery`].
-    pub fn names(&self) -> Vec<String> {
-        self.body.names()
-    }
-
-    /// The fields returned by this [`BoundQuery`].
-    pub fn fields(&self) -> Vec<Field> {
-        self.body.fields()
+    /// The schema returned by this [`BoundQuery`].
+    pub fn schema(&self) -> &Schema {
+        self.body.schema()
     }
 
     /// The types returned by this [`BoundQuery`].
     pub fn data_types(&self) -> Vec<DataType> {
-        self.body.data_types()
+        self.schema().data_types()
     }
```

Comment:
Shall we remove this to be consistent with others?

Reply:
I think it might be ok, because the `BoundQuery` is the top of the Bound tree.

Reply:
Not a big deal. But someone may still add `names` on `BoundQuery` or `data_types`+`names` on `BoundSetExpr`/... later for the same reason: `.data_types()` is shorter than `.schema().data_types()`. Anyways we still achieve the goal on making `select.schema` consistent with `select_exprs` and `aliases`.